Shopify/shopify-cli
# RunExtensionCommand -> ExecuteCommands

### WHY are these changes introduced?

`RunExtensionCommand` was meant to be a light wrapper to run commands for the Extensions Development Server. It is now too coupled, making it hard to see the differences between the commands.

### WHAT is this pull request doing?

This PR splits up `RunExtensionCommand` into a module with a base class, and 3 sub-classes: `Create`, `Build`, and `Serve`. Each class has its own properties to make it easier to understand what is required for each command.

### How to test your changes?

1. within the shopify-cli repo, run rake extensions:install
2. give yourself the required beta flag: shopify config feature extension_server_beta --enable
3. cd into an extension project
4. [CHECKOUT_UI_EXTENSIONS only] run echo "EXTENSION_RESOURCE_URL=/cart/1:1" >> .env in the extension project (this bypasses the call to the AdminAPI to fetch product data to build the resource_url, as this call is intermittently failing for some users)
5. from the extension project, run: `shopify extension build` and `shopify extension serve`

All commands should work as usual.

### Update checklist

~- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)~ not user facing yet
~- [x] I've included any post-release steps in the section above.~ N/A
